packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte

```diff
@@ -309,6 +336,9 @@
       ) {
         await approvers[paymentNetworkExtension.id]();
       }
+
+      const approveStatus = statuses.find((s) => s.name === "APPROVE_ERC20");
```

Comment:
I don't like that we must search the `statuses` array each time we want to look up a state boolean.

Reply:
Me neither, but that is the best solution for now

Reply:
We should be using the index, if we are using the id we will get to the same approach, which is not bad for this small array as I mentioned:
- The find function gives us O(n), which is O(4), very small
- And approach with index is O(1), which is faster, but not noticeable

Reply:
But will add with index search